pkg/modules/authorization.go

```diff
@@ -2232,3 +2265,21 @@ func getRedisChecksumFromSecretData(ctx context.Context, ctrlClient crclient.Cli
 	hash := sha256.Sum256(yamlBytes)
 	return hex.EncodeToString(hash[:]), nil
 }
+
+func mountConfigVolume(authModule csmv1.Module, deployment *appsv1.Deployment) {
+	for _, component := range authModule.Components {
+		for _, config := range component.ConfigSecretProviderClass {
+			if config.ConfigSecretProviderClassName != "" {
```

Comment:
We already extracted the `ConfigSecretProviderClassName` before calling this function. If we can just pass it in, we could avoid looping all the components here. Please see `mountRedisVolumes().

If we are going to refactor this, could you reset the `ConfigSecretProviderClassName` variable to empty. otherwise, the previously extracted value would be used for next deployment.